specs/app.spec.js

```diff
@@ -66,49 +54,104 @@ describe('When the user has csv files in the files folder', function () {
         });
 
         it('Should have inserted the tr csv as expected', function (done) {
-            var connection = mysql.createConnection(DB_CONFIG);
-            connection.query('SELECT * FROM tr', (error, data) => {
-                connection.end();
-                if (error) {
-                    done(error);
-                }
-                expect(data).to.deep.equal([
-                    { '"id"': '1', 'first name': '"SMH"', 'person\'s email': 'smh@email.com' },
-                    { '"id"': '2', 'first name': 'MO\'', 'person\'s email': 'mo@email.com' },
-                    { '"id"': '3', 'first name': 'hello, world', 'person\'s email': 'someemail@email.com' },
-                    { '"id"': '4', 'first name': 'john', 'person\'s email': 'john@email.com' }
-                ]);
-                done();
-            });
+
+            connectAndQuery('SELECT * FROM tr')
+                .then(data => {
+                    expect(data).to.deep.equal([
+                        { '"id"': '1', 'first name': '"SMH"', 'person\'s email': 'smh@email.com' },
+                        { '"id"': '2', 'first name': 'MO\'', 'person\'s email': 'mo@email.com' },
+                        { '"id"': '3', 'first name': 'hello, world', 'person\'s email': 'someemail@email.com' },
+                        { '"id"': '4', 'first name': 'john', 'person\'s email': 'john@email.com' }
+                    ]);
+                    done();
+                });
         });
 
         it('Should have not inserted t1 csv in the files folder', function (done) {
-            var connection = mysql.createConnection(DB_CONFIG);
-            connection.query('SHOW TABLES LIKE \'t1\'', (error, data) => {
-                connection.end();
-                if (error) {
-                    done(error);
-                }
-                expect(data.length).to.deep.equal(0);
-                done();
-            });
+
+            connectAndQuery('SHOW TABLES LIKE \'t1\'')
+                .then(data => {
+                    expect(data.length).to.deep.equal(0);
```

Comment:
`to.deep.equal` is not used when comparing primitives, you would typically use `to.equal` instead. Same comment applies elsewhere.